vlmcsd: add UCI support for vlmcsd

AutoCONFIG opened the pull request:
原本的vlmcsd以及luci-app-vlmcsd比较混乱，特别是init文件和uci支持是通过luci应用来实现的，而非后端程序。
所以重做调整，并使用js重写其luci应用
1. init文件主要源于luci-app-vlmcsd，进行了修改，正在重新修改luci-app-vlmcsd源代码来适配（测试后会提交pr）
2. 删除了0.0.0.0:1688监听选项，因为使用[::]:1688即可完成ipv4和ipv6的双栈监听
3. 添加允许互联网访问的选项，让用户自行决定是否将服务提供到互联网
4. 修改init文件以符合openwrt init文件的开发建议样式

1715173329 commented on net/vlmcsd/files/vlmcsd.init:
```diff
@@ -1,31 +1,65 @@
+
+	if [ "$auto_activate" -eq "1" ];then
+		HOSTNAME=$(uci get system.@system[0].hostname)
+		host_name=$(awk -F ',' '/^[ \t]*srv-host=_vlmcs\._tcp/{print $2}' /etc/dnsmasq.conf)
```
这意味着这个配置只会在启动时加入，而不会被删除。应该尽量减少对 flash 的写入，也尽量减少对这类核心文件的直接修改。
这种情况可以外挂 dnsmasq.d 文件，具体可以参考 unblockneteasemusic 之类的。

1715173329 commented on net/vlmcsd/files/vlmcsd.init:
```diff
@@ -1,31 +1,65 @@
+	procd_append_param command -L "[::]:1688"
+	procd_append_param command -l "/var/log/vlmcsd.log"
+	procd_set_param respawn
+	procd_set_param file /etc/vlmcsd.ini
```
```suggestion
	procd_open_instance "vlmcsd"
	procd_set_param command /usr/bin/vlmcsd
	procd_append_param command -D
	procd_append_param command -i "/etc/vlmcsd.ini"
	procd_append_param command -L "[::]:1688"
	procd_append_param command -l "/var/log/vlmcsd.log"

	procd_set_param file /etc/vlmcsd.ini
	procd_set_param respawn
```

1715173329 commented on net/vlmcsd/files/vlmcsd.init:
```diff
@@ -1,31 +1,65 @@
+	procd_append_param command -D
+	procd_append_param command -i "/etc/vlmcsd.ini"
+	procd_append_param command -L "[::]:1688"
+	procd_append_param command -l "/var/log/vlmcsd.log"
```
对于长久的运行，如果没有 log rotation，可能会出现日志过大导致 oom，这里可以直接写到 syslog 里面。
procd_set_param stdout 1
procd_set_param stderr 1

AutoCONFIG replied:
已做调整，对于dnsmasq操作命令部分，由于只是为了添加简单的srv-host信息，所以使用echo来完全覆盖原本的/tmp/dnsmasq.d/dnsmasq-vlmcsd.conf文件更为合适（假设用户没有重启，存在旧的文件），目标文件没有可以自动创建，文件存在可以完全覆盖掉重新写。

1715173329 replied:
1. 走 syslog 就需要把这里 -l xxx 删掉。
2. 不能再硬编码 /tmp/dnsmasq.d，在 24.10 及新版中，dnsmasq 会给 include 文件夹加上 uciconf 后缀，这部分可以参考 unblockneteasemusic。
3. 你的 dnsmasq 配置文件从来没有回收过，需要在停止时移除。

1715173329 commented on net/vlmcsd/files/vlmcsd.init:
```diff
@@ -1,31 +1,63 @@
 #!/bin/sh /etc/rc.common
-# Copyright (c) 2011-2015 OpenWrt.org
+# Copyright (c) 2011-2025 OpenWrt.org
```
如果不是该版权所有者，就不要动这部分。
鉴于该脚本已经是处于重写状态，可以替换成你自己的。

1715173329 commented on net/vlmcsd/files/vlmcsd.init:
```diff
@@ -1,31 +1,79 @@
+	config_get_bool internet_access "config" "internet_access" "0"
+
+	if [ "$enabled" -eq "0" ];then
+		if [ -f "$DNSMASQ_VLMCSD" ];then
```
这里没必要，可以用
```
reload_service() {
	stop
	start
}
```

这样 reload 的时候就会执行 stop_service 函数。

1715173329 commented on net/vlmcsd/files/vlmcsd.init:
```diff
@@ -1,31 +1,79 @@
-		echo "KMS Server has stopped."
+	if [ "$auto_activate" -eq "1" ];then
+		HOSTNAME=$(uci get system.@system[0].hostname)
+		echo "srv-host=_vlmcs._tcp,$HOSTNAME,1688,0,100" > $DNSMASQ_VLMCSD
```
```suggestion
		hostname="$(uci -q get system.@system[0].hostname)"
		echo "srv-host=_vlmcs._tcp,$hostname,1688,0,100" > "$DNSMASQ_VLMCSD"
```

非全局变量建议用小写。

AutoCONFIG replied:
感谢天灵老师。已经完成修改

1715173329 commented on net/vlmcsd/files/vlmcsd.init:
```diff
@@ -1,31 +1,78 @@
+	[ "$enabled" -eq "1" ] || return 1
+
+	if [ "$auto_activate" -eq "1" ];then
+		hostname=$(uci get system.@system[0].hostname)
```
```suggestion
		hostname="$(uci get system.@system[0].hostname)"
```

根据 [SC2086](https://www.shellcheck.net/wiki/SC2086)，请在这里加上引号，其余 LGTM。

AutoCONFIG replied:
其余部分已经检查，并通过shellcheck工具检查

AutoCONFIG committed:
vlmcsd: add UCI support for vlmcsd

- Derived from luci-app-vlmcsd with some modifications. The source code
  of luci-app-vlmcsd needs to be re-modified for adaptation.
- Removed the listening option for 0.0.0.0:1688 because [::]:1688 can
  achieve dual-stack listening for both IPv4 and IPv6.
- Added an option to allow Internet access, enabling users to decide
  whether to expose the service to the Internet on their own.
- Modified the init file to meet the standards of OpenWrt init files.

Signed-off-by: Yun Wang <maoerpet@foxmail.com>
